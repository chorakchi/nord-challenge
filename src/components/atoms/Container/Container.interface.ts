import { HTMLAttributes } from "react";

export interface TypographyProps extends HTMLAttributes<HTMLElement> {}
