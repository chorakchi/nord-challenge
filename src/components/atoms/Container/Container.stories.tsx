import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Container } from "./Container";

export default {
  title: "Atoms/Container",
  component: Container,
  parameters: {
    docs: {
      description: {
        component: `While containers can be nested, most layouts do not require a nested container.`,
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Container>;

const Template: ComponentStory<typeof Container> = (args) => (
  <Container {...args} />
);

export const Container_ = Template.bind({});
Container_.args = {
  children: "Lorem Ipsum",
};
