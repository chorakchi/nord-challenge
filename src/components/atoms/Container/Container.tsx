import { TypographyProps } from "./Container.interface";
export const Container: React.FC<TypographyProps> = ({
  children,
  ...props
}: TypographyProps) => {
  return (
    <div
      {...props}
      className="flex flex-col h-screen container mx-auto justify-center"
    >
      <div className="fade-in min-h-full flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
        {children}
      </div>
    </div>
  );
};
