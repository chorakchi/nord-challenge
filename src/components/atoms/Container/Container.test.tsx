import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Container } from "./Container";

describe("Container", () => {
  it("should render the Container", () => {
    const subject = shallow(<Container>Container</Container>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Container>Container</Container>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
