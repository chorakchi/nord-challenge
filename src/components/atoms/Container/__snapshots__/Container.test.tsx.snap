// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render correctly and match to the Snapshot 1`] = `
<div
  className="flex flex-col h-screen container mx-auto justify-center"
>
  <div
    className="fade-in min-h-full flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8"
  >
    Container
  </div>
</div>
`;
